com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -267,7 +267,10 @@ public void LazyInitialize()
             Academy.Instance.AgentForceReset += _AgentReset;
             m_Brain = m_PolicyFactory.GeneratePolicy(Heuristic);
             ResetData();
+#pragma warning disable 0618
```

Comment:
This is a warning about calling an obsolete method, right?

Would it be cleaner to only call the new version here, and make the default implementation of `Initialize` call `InitializeAgent` (wrapped in the pragmas)? That way there's no chance that someone accidentally implements both methods and we call them both

Reply:
(same comments for AgentReset and AgentAction too)